samples/net/cloud/mqtt_azure/src/main.c

```diff
@@ -249,8 +249,9 @@ static void mqtt_event_handler(struct mqtt_client *const client,
 					data,
 					len >= sizeof(data) - 1 ?
 					sizeof(data) - 1 : len);
-			if (bytes_read < 0 && bytes_read != -EAGAIN) {
-				LOG_ERR("failure to read payload");
+			if (bytes_read < 0) {
```

Comment:
I think `mqtt_read_publish_payload_blocking` should be used here instead. `mqtt_read_publish_payload` is non-blocking function, so returning `-EAGAIN` is not really an error (but the design of this loop isn't great since it'll keep spinning in that case apart from the issue detected by coverity). With the blocking function, it should be safe to abort on any error.

Reply:
That sounds better indeed. I am not able to test these kind of changes right now so need to postpone this PR a bit.